### ISSUE TITLE
Client event routing issue

### ISSUE THREAD

**Issue by MathieuBerthellemy** [ldapjs/node-ldapjs#945]
File `./lib/client/client.js` Line 1260

```
let event = msg.constructor.name
// Generate the event name for the event emitter, i.e. "searchEntry"
// and "searchReference".
event = (event[0].toLowerCase() + event.slice(1)).replaceAll('Result', '')
return sendResult(event, msg)
```

For some reason event[0] is undefined and therefore my app crashes on the _toLowerCase_ call

I suggest to replace the if block with:

```
if(msg instanceof SearchEntry) {
      return sendResult(event, 'searchEntry')
} else if(msg instanceof SearchReference) {
      return sendResult(event, 'searchReference')
}
```

I think it's a terrible idea to base decision on constructor.name value.
I've struggled 2.5 days to figure it out.

**Comment by jsumners** [comment]
Please provide a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example). Doing so will help us diagnose your issue. It should be the bare minimum code needed to trigger the issue, and easily runnable without any changes or extra code.

You may use a GitHub repository to host the code if it is too much to fit in a [code block](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) (or two).

**Comment by melissakintz** [comment]
Hello, same error for me and [here](https://stackoverflow.com/questions/77190681/cannot-perform-ldap-search-operation-ldapjs-osixia-openldap)

**Comment by MathieuBerthellemy** [comment]
@melissakintz yes that's actually a post of mine, I'm gonna mark it as resolved and link the post to this ticket

**Comment by DukeVenator** [comment]
We had the same problem and applied the patch mentioned above using a nifty NPM module called [patch-package](https://www.npmjs.com/package/patch-package) to persist it across our development team.




`if(msg instanceof SearchEntry) {
      return sendResult('searchEntry', msg)
} else if(msg instanceof SearchReference) {
      return sendResult('searchReference', msg)`


This package unmodified works on Node18 on Windows but when we deploy to our node image running on ocp 3.11 on a container the issue the OP occurs. 

By applying the fix our NextAuth works flawlessly.  I suggest perhaps taking a closer look at this issue.
